src/c2goto/library/string.c

```diff
@@ -266,16 +269,39 @@ __ESBMC_HIDE:;
   return cpy;
 }
 
-void *memcpy(void *dst, const void *src, size_t n)
+void *__memcpy_impl(void *dst, const void *src, size_t n)
 {
 __ESBMC_HIDE:;
+  if (n == 0)
+    return dst;
+
   char *cdst = dst;
   const char *csrc = src;
+
   for (size_t i = 0; i < n; i++)
     cdst[i] = csrc[i];
+
   return dst;
 }
 
+void *memcpy(void *dst, const void *src, size_t n)
+{
+__ESBMC_HIDE:;
+  if (n == 0 || src == NULL || dst == NULL)
+    return dst;
+
+  // char *cdst = dst;
+  // const char *csrc = src;
+
+  //for (size_t i = 0; i < n; i++)
+  // cdst[i] = csrc[i];
+
+  void *hax = &__memcpy_impl;
+  (void)hax;
+
+  return __ESBMC_intrinsic_memcpy(dst, src, n);
+}
```

Comment:
Could you follow the same pattern as with memset please?
That is, memcpy just calls `__ESBMC_memcpy` which will than either be intrinsified or fallback to `__memcpy_impl`?